internal/state/graph/gateway_listener.go

```diff
@@ -382,20 +376,42 @@ func createPortConflictResolver() listenerConflictResolver {
 func createExternalReferencesForTLSSecretsResolver(
 	gwNs string,
 	secretMemoryMgr secrets.SecretDiskMemoryManager,
+	refGrants map[types.NamespacedName]*v1beta1.ReferenceGrant,
 ) listenerExternalReferenceResolver {
 	return func(l *Listener) {
-		nsname := types.NamespacedName{
-			Namespace: gwNs,
-			Name:      string(l.Source.TLS.CertificateRefs[0].Name),
+		certRef := l.Source.TLS.CertificateRefs[0]
```

Comment:
Do we need to verify that TLS is not nil before we access it?

Reply:
We check for nil here: https://github.com/nginxinc/nginx-kubernetes-gateway/blob/e04cc279bcdd70ce31996908be47d970a898e085/internal/state/graph/gateway_listener.go#L268

Do you think a comment about that would help here?

Reply:
Nope, I'm good with it as-is.